File: IagoAuction/ClientApp/src/app/shared/models/lot.ts
import { Painting } from "./painting";
import { Bid } from "./bid";

export class Lot {
  id: number;
  painting: Painting;
  bids: Bid[];

  constructor(id: number, painting: Painting, bids: Bid[]) {
    this.id = id;
    this.painting = painting;
    this.bids = bids != null ? bids : [];
  }
}

File: IagoAuction/Models/Lot.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Mvc.Formatters.Xml;

namespace IagoAuction.Models
{
    public class Lot
    {
        public int Id { get; set; }

        public List<Bid> Bids { get; set; }

        public int? PaintingId { get; set; }
        [ForeignKey("paintingId")]
        public Painting Painting { get; set; }

        public int? AuctionId { get; set; }
        [ForeignKey("AuctionId")]
        public Auction Auction { get; set; }

        public Lot()
        {
            this.Bids = new List<Bid>();
        }
    }
}
